From 72869da97db070bc00cc0934aebdb8f247b05b55 Mon Sep 17 00:00:00 2001
From: Jiffin Tony Thottan <jthottan@redhat.com>
Date: Thu, 2 Mar 2017 12:22:30 +0530
Subject: [PATCH 56/74] ganesha/scripts : Stop ganesha process on all nodes if
 cluster setup fails

During staging phase of volume option "nfs-ganesha", symlink "ganesha.conf"
will be created plus ganesha process will be started. The cluster setup
happens during commit phase of that option. So if cluster set up fails, the
ganesha process will be running on all cluster nodes.

Change-Id: Ib2cb85364b7ef5b702acb4826ffdf8e6f31a2acd
Signed-off-by: Jiffin Tony Thottan <jthottan@redhat.com>
Reviewed-on: https://review.gluster.org/16823
Smoke: Gluster Build System <jenkins@build.gluster.org>
Tested-by: Kaleb KEITHLEY <kkeithle@redhat.com>
Reviewed-by: soumya k <skoduri@redhat.com>
Reviewed-by: Kaleb KEITHLEY <kkeithle@redhat.com>
NetBSD-regression: NetBSD Build System <jenkins@build.gluster.org>
CentOS-regression: Gluster Build System <jenkins@build.gluster.org>
---
 extras/ganesha/scripts/ganesha-ha.sh | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/extras/ganesha/scripts/ganesha-ha.sh b/extras/ganesha/scripts/ganesha-ha.sh
index 2d6b06f..df4f0b8 100644
--- a/extras/ganesha/scripts/ganesha-ha.sh
+++ b/extras/ganesha/scripts/ganesha-ha.sh
@@ -175,6 +175,13 @@ determine_servers()
     fi
 }
 
+stop_ganesha_all()
+{
+    local serverlist=${1}
+    for node in ${serverlist} ; do
+        manage_service "stop" ${node}
+    done
+}
 
 setup_cluster()
 {
@@ -191,6 +198,8 @@ setup_cluster()
     pcs cluster setup ${RHEL6_PCS_CNAME_OPTION} ${name} --transport udpu ${servers}
     if [ $? -ne 0 ]; then
         logger "pcs cluster setup ${RHEL6_PCS_CNAME_OPTION} ${name} ${servers} failed"
+        #set up failed stop all ganesha process and clean up symlinks in cluster
+        stop_ganesha_all ${servers}
         exit 1;
     fi
     pcs cluster start --all
-- 
1.8.3.1

